### PR TITLE
Precompute Trig functions in Indexing

### LINKED ISSUE #20150: Indexing Utils Performance
A quick run of the profiler while running `FindPeaksUsingLatticeParameterTest` shows that the majority of the time is spent calculating `cos` and `sin` functions. These can be pre-computed as they are based on fixed lattice angles.

`IndexingUtils` is still a mess. It could do with some attention and porting to `Eigen`. This issue will not resolve those major issues.

### PR DESCRIPTION
**Pull request by OwenArnold**
Unnecessary re-computation in UB scanning

**Tester** 
* All tests should pass.
* This should be faster than what we originally had.

fixes #20150

### CODE REVIEW

**Review by VickieLynch — APPROVED** [review]
Same answer, but faster with TOPAZ data.

File: Framework/Geometry/inc/MantidGeometry/Crystal/IndexingUtils.h [hunk 1]
```diff
@@ -136,9 +136,10 @@ class MANTID_GEOMETRY_DLL IndexingUtils {
                           double req_tolerance, double min_vol);
 
   /// Get the vector in the direction of "c" given other unit cell information
-  static Kernel::V3D Make_c_dir(const Kernel::V3D &a_dir,
-                                const Kernel::V3D &b_dir, double c,
-                                double alpha, double beta, double gamma);
+  static Kernel::V3D makeCDir(const Kernel::V3D &a_dir,
+                              const Kernel::V3D &b_dir, const double &c,
+                              const double &cosAlpha, const double &cosBeta,
+                              const double &cosGamma, const double &sinGamma);
```
Comment (OwenArnold) [inline Framework/Geometry/inc/MantidGeometry/Crystal/IndexingUtils.h]: Absolutely none. Yes, i'll make that change.

**Comment by OwenArnold** [comment]
@VickieLynch I wanted to speed up the total system unit test as it's the longest running in the whole framework (FindUBUsingLatticeParametersTest). The majority of the time is now spent loading the TOPAZ peaks file. It would be really nice to replace this with something faster, or some other example, ideally made-in-memory data set. Do you have any ideas on how this could be done?

**Comment by VickieLynch** [comment]
@OwenArnold LoadIsawPeaks was made to run very fast by someone in the UK in the past, but recently it is very slow (because of instrument 2.0 changes?) even for small files.  I would prefer it be improved rather that the test made smaller.

**Comment by VickieLynch** [comment]
@OwenArnold Mike Hart Improve performance on LoadIsawPeaks algorithm #13961

**Comment by OwenArnold** [comment]
Thanks @VickieLynch. I'll take a look to see what's making this slower.

**Pull request merged**